google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITQueryWatchTest.java

```diff
@@ -0,0 +1,484 @@
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import javax.annotation.Nullable;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestName;
+
+public final class ITQueryWatchTest {
+
+  private static Firestore firestore;
+
+  @Rule public TestName testName = new TestName();
+
+  private CollectionReference randomColl;
+
+  @BeforeClass
+  public static void beforeClass() {
+    FirestoreOptions firestoreOptions = FirestoreOptions.newBuilder().build();
+    firestore = firestoreOptions.getService();
+  }
+
+  @Before
+  public void before() {
+    String autoId = LocalFirestoreHelper.autoId();
+    String collPath = String.format("java-%s-%s", testName.getMethodName(), autoId);
+    randomColl = firestore.collection(collPath);
+  }
+
+  @AfterClass
+  public static void afterClass() throws Exception {
+    firestore.close();
+  }
+
+  /*
+  Start a listener on a query with an empty result set
+   */
+  @Test
+  public void emptyResults() throws InterruptedException {
+    final Query query = randomColl.whereEqualTo("foo", "bar");
+    // register the snapshot listener for the query
+    QuerySnapshotEventListener listener = new QuerySnapshotEventListener(1);
+    ListenerRegistration registration = query.addSnapshotListener(listener);
+
+    try {
+      listener.eventsCDL.await();
+    } finally {
+      registration.remove();
+    }
+
+    assertThat(listener.receivedEvents).hasSize(1);
+    ListenerEvent e = listener.receivedEvents.get(0);
+    assertThat(e.error).isNull();
+    QuerySnapshot snap = e.value;
+    assertThat(snap).isNotNull();
+    assertThat(snap).isEmpty();
+    assertThat(snap.getDocumentChanges()).isEmpty();
+  }
+
+  /*
+  Start a listener on a query with a non-empty result set
+   */
+  @Test
+  public void nonEmptyResults() throws InterruptedException, TimeoutException, ExecutionException {
+    // create a document in our collection that will match the query
+    randomColl.document("doc").set(map("foo", "bar")).get(5, TimeUnit.SECONDS);
+
+    // run our test
+    final Query query = randomColl.whereEqualTo("foo", "bar");
+    QuerySnapshotEventListener listener = new QuerySnapshotEventListener(1);
+    List<ListenerEvent> receivedEvents = listener.receivedEvents;
+    ListenerRegistration registration = query.addSnapshotListener(listener);
+
+    try {
+      listener.eventsCDL.await();
+    } finally {
+      registration.remove();
+    }
+
+    assertThat(receivedEvents).hasSize(1);
+    ListenerEvent e = receivedEvents.get(0);
+    assertThat(e.error).isNull();
+    QuerySnapshot snap = e.value;
+    assertThat(snap).isNotNull();
+    assertThat(snap).hasSize(1);
+  }
+
+  /*
+  Starting from a query with empty results creating a new document
+  that matches the query should result in an ADDED event
+   */
+  @Test
+  public void emptyResults_newDocument_ADDED()
```

Comment:
This name could use a sprinkle of the Java Style Guide :)

Reply:
This type of scenario is specifically called out in the style guide: http://go/java-style#s5.2.3-method-names
> Underscores may appear in JUnit test method names to separate logical components of the name, with each component written in lowerCamelCase

Is your ask here for me to not use the upper enum names? i.e. `s/ADDED/added/`?